ios/Exponent/Kernel/ReactAppManager/EXReactAppManager.mm

```diff
@@ -113,9 +113,18 @@ - (void)rebuildBridge
 {
   EXAssertMainThread();
   NSAssert((_delegate != nil), @"Cannot init react app without EXReactAppManagerDelegate");
+
   [self _invalidateAndClearDelegate:NO];
   [self computeVersionSymbolPrefix];
 
+  // Assert early so we can catch the error before instantiating the bridge, otherwise we would be passing a
+  // nullish scope key to the scoped modules.
+  // Alternatively we could skip instantiating the scoped modules but then singletons like the one used in
+  // expo-updates would be loaded as bare modules. In the case of expo-updates, this would throw a fatal error
+  // because Expo.plist is not available in the Expo Go app.
+  NSAssert(_appRecord.scopeKey, @"Experience scope key should be nonnull when getting initial properties for root view. This can occur when the manifest JSON, loaded from the server, is missing keys.");
```

Comment:
This one is actually n/a given the more recent changes, but I think we should keep it since it makes debugging easier.

Reply:
@wschurman does this seem OK? IIRC we talked about making scopeKey optional since standalone apps won't need it one day.

(An argument for why this is OK to keep here is that this code is only in Expo Go and I think classic builds. Expo Go will always need scope keys, and modern builds e.g. made with EAS Build won't include this code anyway.)

Reply:
I think this should be fine. There are a lot of places in Expo Go and standalone clients that assume it exists.

I don't know that we'll ever be able to get rid of scopeKey fully. Copied from my notion comment:

> In an ideal world we wouldn't use scopeKey in production but as far as I can tell it may not actually be possible to move off of it. There is code that we have in place that "unscopes" stored data due to us mistakenly scoping data in production apps, but the issue is that unscoping the data requires the scope key, so as long as we have that unscoping code we need to verify scope key in production.